services/MerchStoreService.ts

```diff
@@ -208,6 +212,33 @@ export default class MerchStoreService {
         });
       }
 
+      if (photos) {
+        // error on duplicate photo uuids
+        const dupSet = new Set();
+        photos.forEach((photo) => {
+          if (dupSet.has(photo.uuid)) { throw new UserError(`Multiple edits is made to photo: ${photo.uuid}`); }
+          dupSet.add(photo.uuid);
+        });
+
```

Comment:
To make life much much simpler, I don't think we should support rearranging photos and their indexes for an item. See our Discord discussion [here](https://discord.com/channels/573028991527550986/791028414538186772/1145092442392105150) - needing to deal with rearranging photos and updating indexes (as you can see with this function's implementation) is not clean, not efficient, and not worth the gains we get from supporting this feature (because admins can instead just delete/add photos to modify the list as needed). So, let's remove this function's photo implementation entirely.

Reply:
In fact, let's not worry about photo position at all in the backend. Have a look at how we implemented MerchItemOptionMetadata - it follows the exact same pattern we're looking for here (can remove any option metadata, but can only add to the end), but doesn't have *any* logic around positions that takes place in the backend. This is intentional, so that the frontend can have full agency over how they want to implement positioning (e.g. if they want to allow pre-pending photos to the front, they can use negative positions). The backend's job is to just persist that representation.

Reply:
Sure I will make it consistent